### PR TITLE
K8SPG-378: Add crVersion field

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPG-378](https://jira.percona.com/browse/K8SPG-378) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
No crVersion field.

**Solution:**
Add crVersion field and label the resources with it.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are the manifests (crd/bundle) regenerated if needed?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PG version?
- [x] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

File: percona/controller/pgcluster/controller_test.go [hunk 1]
```diff
@@ -701,3 +701,91 @@ var _ = Describe("Users", Ordered, func() {
...
+			"postgres-operator.crunchydata.com/instance-set": "instance1",
+			"postgres-operator.crunchydata.com/cluster":      crName,
+		}
+		err = k8sClient.List(ctx, stsList, client.InNamespace(cr.Namespace), client.MatchingLabels(labels))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(stsList.Items).NotTo(BeEmpty())
+
+		for _, sts := range stsList.Items {
+			Expect(sts.Labels).To(HaveKeyWithValue(v2beta1.LabelOperatorVersion, cr.Spec.CRVersion))
+		}
```
Comment (inelpandzic) [inline percona/controller/pgcluster/controller_test.go:759]: A bit more idiomatic assertion here would be:
```
Expect(stsList.Items).Should(ContainElement(gs.MatchFields(gs.IgnoreExtras, gs.Fields{
	"ObjectMeta": gs.MatchFields(gs.IgnoreExtras, gs.Fields{
		"Labels": HaveKeyWithValue(v2beta1.LabelOperatorVersion, cr.Spec.CRVersion),
	}),
})))
```
The same goes for other two places like this.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/293c993de/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/469/commits/293c993de5e3846df1d143139934d4b9d40c7b16
image: `perconalab/percona-postgresql-operator:PR-469-293c993de`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/e8e8e62d7/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/469/commits/e8e8e62d718f450e05f07a88e6d0b070e80bbc35
image: `perconalab/percona-postgresql-operator:PR-469-e8e8e62d7`

File: percona/version/version.go [hunk 2]
```diff
@@ -71,7 +66,7 @@ func fetchVersions(ctx context.Context, endpoint string, vm Meta) (DepVersion, e
 		HelmDeployCr:       &vm.HelmDeployCR,
 		HelmDeployOperator: &vm.HelmDeployOperator,
 		SidecarsUsed:       &vm.SidecarsUsed,
-		Product:            ProductName,
+		Product:            v2beta1.ProductName,
```
Comment (hors) [inline percona/version/version.go:69]: I think it will be v2. @egegunes should we wait and then update it?

**Review by hors — CHANGES_REQUESTED** [review]

File: percona/version/version.go (hunk 2 above)
Comment (inelpandzic) [reply percona/version/version.go:69]: Yes, we should.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/11428a6b4/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/469/commits/11428a6b4eec36c468c925882117f809bb81671f
image: `perconalab/percona-postgresql-operator:PR-469-11428a6b4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-469/4d309a5c5/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/469/commits/4d309a5c54a7ba8bf07ca2a45b5d661266b0cbaf
image: `perconalab/percona-postgresql-operator:PR-469-4d309a5c5`

**Review by hors — APPROVED** [review]

**Pull request merged**